File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char a) {
        return (a == 'a' || a == 'e' || a == 'i' || a == 'o' || a == 'u' ||
                a == 'A' || a == 'E' || a == 'I' || a == 'O' || a == 'U');
    }
    string reverseVowels(string s) {
        int n = s.size();
        // s.toLowerCase();
        for (int i = 0, j = n - 1; i < j;) {
            if (!isVowel(s[i])) {
                i++;
            }
            if (!isVowel(s[j])) {
                j--;
            }
            if (isVowel(s[i]) && isVowel(s[j])) {
                swap(s[i], s[j]);
                i++;
                j--;
            }
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public boolean isVowel(char ch)
    {
        return (ch == 'a' || ch == 'e' || ch =='i'||ch=='o'||ch=='u'
               ||ch=='A'||ch=='E'||ch=='I'||ch=='O' ||ch=='U');
    }
    public String reverseVowels(String s) {
        // Write your code here
        int n = s.length();
        char[] arr = s.toCharArray();
        int i=0,j=n-1;
        
        while(i<j)
        {
            while(i<j && !isVowel(arr[i]))  i++;
            while(i<j && !isVowel(arr[j]))  j--;
            
            char temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            
            i++;
            j--;
        }
        return new String(arr);
    }
}